### PR TITLE
[PERF-231] Make calls async at the resource level

### PR DESCRIPTION
**Pull request by bradbanister**

### CODE REVIEW

File: src/edfi-paging-test/edfi_paging_test/performance_tester.py [hunk 1]
```diff
@@ -52,16 +56,28 @@ def fetch_resource(request_client: RequestClient, target_resource: str) -> None:
...
+        loop: asyncio.AbstractEventLoop = asyncio.get_event_loop()
+
+        fetch_resource_calls: List[asyncio.Future] = [
+            loop.run_in_executor(
+                executor, fetch_resource, request_client, target_resource
+            )
+            for target_resource in args.resourceList
+        ]
+
+        await asyncio.wait(fetch_resource_calls)
```
Comment (bradbanister) [inline src/edfi-paging-test/edfi_paging_test/performance_tester.py:76]: A lot like Promise.allSettled() in JS

File: src/edfi-paging-test/edfi_paging_test/performance_tester.py [hunk 2]
```diff
@@ -52,16 +56,28 @@ def fetch_resource(request_client: RequestClient, target_resource: str) -> None:
...
+async def run(args: MainArguments) -> None:
 
     try:
         logger.info("Starting paging volume test...")
+        start = time.time()
         request_client: RequestClient = RequestClient(args)
 
-        for target_resource in args.resourceList:
-            fetch_resource(request_client, target_resource)
+        executor: ThreadPoolExecutor = ThreadPoolExecutor(max_workers=5)
```
[the remark this answers is not in the document]
Comment (stephenfuqua) [reply src/edfi-paging-test/edfi_paging_test/performance_tester.py:66]: We have a command line argument for that - `connectionLimit`.

**Comment by stephenfuqua** [comment]
Nice improvement. Tested with 5 executions each for the matrix before/after and page size 20/100. Also confirmed that we have the expected output after the update. 

image

This test was using the production API sandbox, populated template, with these resources: `"Students", "StudentSchoolAssociations", "tpdm/candidates", "StudentEducationOrganizationAssociations"`

**Review by stephenfuqua — CHANGES_REQUESTED** [review]
Looks good, let's just add the connection limit from the configuraiton.

File: src/edfi-paging-test/edfi_paging_test/helpers/argparser.py [hunk 3]
```diff
@@ -54,7 +54,7 @@ def parse_main_arguments() -> MainArguments:
         "--connectionLimit",
         help="Maximum concurrent connections to api",
         type=int,
-        default=4,
+        default=5,
```
Comment (stephenfuqua) [inline src/edfi-paging-test/edfi_paging_test/helpers/argparser.py:57]: There's definitely some overhead. I tried switching to 10 and it got much worse. 7 was a little better, 8 got worse again. Mileage may vary. 5 seems like a good default.

**Review by stephenfuqua — APPROVED** [review]

**Pull request merged**